docs/architecture/cloud-native/azure-security.md

```diff
@@ -0,0 +1,257 @@
+
+Setting up this tier of encryption requires running through a wizard in SQL Server Management Studio to select the sort of encryption and where in Key Vault to store the associated keys. 
+
+![Figure 10-5 Selecting columns in a table to be encrypted using Always Encrypted](media/always-encrypted.png)
+
+Client applications that read information from these encrypted columns need to make special allowances to read encrypted data. Connection strings need to be updated with `Column Encryption Setting=Enabled` and client credentials must be retrieved from the Key Vault. The SQL Server client must then be primed with the column encryption keys. Once that is done, the remaining actions use the standard interfaces to SQL Client. That is, tools like Dapper and Entity Framework, which are built on top of SQL Client, will continue to work without changes. Always Encrypted may not yet be available for every SQL Server driver on every language.
+
+The combination of TDE and Always Encrypted, both of which can be used with client-specific keys, ensures that even the most exacting encryption requirements are supported.
+
+### Cosmos DB
+
+Cosmos DB is the newest database provided by Microsoft in Azure. It has been built from the ground up with security and cryptography in mind. AES-256bit encryption is standard for all Cosmos DB databases and can't be disabled. Coupled with the TLS 1.2 requirement for communication, the entire storage solution is encrypted.
+
+![Figure 10-6 The flow of data encryption within Cosmos DB](media/cosmos-encryption.png)
+
+While Cosmos DB doesn't provide for supplying customer encryption keys, there has been significant work done by the team to ensure it remains PCI-DSS compliant without that. Cosmos DB also doesn't support any sort of single column encryption similar to Azure SQL's Always Encrypted yet.
+
+## Keeping secure
+
+Azure has all the tools necessary to release a highly secure product. However, a ladder is only as strong as its weakest rung. If the applications deployed on top of Azure aren't developed with a proper security mindset and good security audits, then they become the weak rung, the weak link in the chain, the unlocked door. There are many great [static analysis tools](https://www.whitesourcesoftware.com/), [encryption libraries](https://www.libressl.org/), and [security practices](https://azure.microsoft.com/resources/videos/red-vs-blue-internal-security-penetration-testing-of-microsoft-azure/) that can be used to ensure that the software installed on Azure is as secure as Azure itself.  
```

Comment:
never heard of weak rung. Also do you need so many analogies here?

Instead of linking the examples directly from the terms, wouldn't be better to say something like:

```markdown
There are many great static analysis tools, encryption libraries, and security practices that can be used to ensure that the software installed on Azure is as secure as Azure itself. [WhiteSource]([static analysis tools](https://www.whitesourcesoftware.com/), [encryption libraries](https://www.libressl.org/), and [security practices](https://azure.microsoft.com/resources/videos/red-vs-blue-internal-security-penetration-testing-of-microsoft-azure/), LibreSSL(https://www.libressl.org/) and [Red vs. Blue - Internal security penetration testing of Microsoft Azure](https://azure.microsoft.com/en-us/resources/videos/red-vs-blue-internal-security-penetration-testing-of-microsoft-azure/) are examples of that, respectively.
```

Reply:
Yeah I considering changing this to chain->weakest link. Weakest rung was new to me, too.

I'm fine with your change, too, regarding the examples.